qt/python/mantidqt/mantidqt/widgets/sliceviewer/cutviewer/view.py

```diff
@@ -0,0 +1,317 @@
+        for icol in range(self.table.columnCount()):
+            for irow in range(self.table.rowCount()):
+                item = QTableWidgetItem()
+                if icol == 5:
+                    item.setData(Qt.EditRole, int(1))
+                else:
+                    item.setData(Qt.EditRole, float(1))
+                if irow == self.table.rowCount()-1:
+                    item.setFlags(item.flags() ^ Qt.ItemIsEditable)  # disable editing in last row (out of plane dim)
+                    item.setBackground(QColor(250, 250, 250))
+                else:
+                    item.setFlags(item.flags() | Qt.ItemIsEditable)
+                self.table.setItem(irow, icol, item)
+
+    def _format_cut_figure(self):
+        self.figure.axes[0].ignore_existing_data_limits = True
+        self.figure.axes[0].autoscale_view()
+        self._format_cut_xabel()
+        for textobj in self.figure.findobj(text.Text):
+            textobj.set_fontsize(8)
+        self.figure.tight_layout()
+
+    def _format_cut_xabel(self):
+        xlab = self.figure.axes[0].get_xlabel()
+        istart = xlab.index('(')
+        iend = xlab.index(')')
+        xunit_str = xlab[iend + 1:].replace('Ang^-1', '$\\AA^{-1}$)').replace('(', '').replace(')', '')
+        xlab = xlab[0:istart] + xlab[istart:iend + 1].replace(' ', ', ') + xunit_str
+        self.figure.axes[0].set_xlabel(xlab)
+
+
+class CutRepresentation:
+    def __init__(self, canvas, notify_on_release, xmin, xmax, ymin, ymax, thickness):
+        self.notify_on_release = notify_on_release
+        self.canvas = canvas
+        self.ax = canvas.figure.axes[0]
+        self.thickness = thickness
+        self.start = self.ax.plot(xmin, ymin, 'ow', label='start')[0]
+        self.end = self.ax.plot(xmax, ymax, 'ow', label='end')[0]
+        self.line = None
+        self.mid = None
+        self.box = None
+        self.mid_box_top = None
+        self.mid_box_bot = None
+        self.current_artist = None
+        self.cid_release = self.canvas.mpl_connect('button_release_event', self.on_release)
+        self.cid_press = self.canvas.mpl_connect('button_press_event', self.on_press)
+        self.cid_motion = self.canvas.mpl_connect('motion_notify_event', self.on_motion)
+        self.draw()
+
+    def remove(self):
+        self.clear_lines(all_lines=True)
+        for cid in [self.cid_release, self.cid_press, self.cid_motion]:
+            self.canvas.mpl_disconnect(cid)
+
+    def draw(self):
+        self.draw_line()
+        self.draw_box()
+        self.canvas.draw()
+
+    def get_start_end_points(self):
+        xmin, xmax = self.start.get_xdata()[0], self.end.get_xdata()[0]
+        ymin, ymax = self.start.get_ydata()[0], self.end.get_ydata()[0]
+        return xmin, xmax, ymin, ymax
+
+    def get_mid_point(self):
+        xmin, xmax, ymin, ymax = self.get_start_end_points()
+        return 0.5 * (xmin + xmax), 0.5 * (ymin + ymax)
+
+    def get_perp_dir(self):
```

Comment:
Please can you add some tests for some of the functions on this class (get_perp_dir\get_mid_point, get_start_end_points)